### PR TITLE
update kernel and build system

### PR DESCRIPTION
**Pull request by magnetophon**
The first commit is already merged.

I have built and run 4.14.18-rt15, but have not tested building the other kernels.

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/linux-3.18-rt.nix [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-import <nixpkgs/pkgs/os-specific/linux/kernel/generic.nix> (args // rec {
+with stdenv.lib;
+
+buildLinux (args // rec {
   kversion = "3.18.51";
   pversion = "rt57";
   version = "${kversion}-${pversion}";
-  extraMeta.branch = "3.18";
+    # branchVersion needs to be x.y
+  extraMeta.branch = concatStrings (intersperse "." (take 2 (splitString "." version)));
```
[the remark this answers is not in the document]
Comment (magnetophon) [reply pkgs/os-specific/linux/kernel/linux-3.18-rt.nix]: Yah, I got it from upstream, so maybe that tickles your "keep it simple"-gland ;)

[the remark this answers is not in the document]
Comment (henrytill) [reply pkgs/os-specific/linux/kernel/linux-3.18-rt.nix]: That's a good enough justification for me!  

We can always do:
```
extraMeta.branch = with stdenv.lib; concatStrings ...
```

**Comment by magnetophon** [comment]
Any news on merging of your work?
Then I can get on with mine: When  I update I often need to build a kernel anyway, so I might as well build the latest.  :)

**Comment by henrytill** [comment]
Oh, whoops - I was waiting for the thumbs up from you.  I'll go ahead and do that.

**Comment by magnetophon** [comment]
> I was waiting for the thumbs up from you.

While a very nice gesture, there's no need to ever do that.

You are far more competent when it comes to nix, and while I may have started this project, you have contributed so much more, that I think you can fully consider it your project.

**Pull request closed without merging**